File: GdPicture/WinForm/C#/Image Processing/frmOpen.cs
using GdPicture14;
using System;
using System.IO;
using System.Windows.Forms;

namespace Image_Processing
{
    public partial class frmOpen
    {

        private readonly GdViewer _owner;

        public frmOpen(GdViewer owner)
        {
            InitializeComponent();
            _owner = owner;
        }

        private void UpdateUi()
        {
            if (rbOpenFromFile.Checked)
            {
                pOpenFromFile.Enabled = true;
                pOpenFromHTTP.Enabled = false;
                pOpenFromFTP.Enabled = false;
            }
            else if (rbOpenFromHTTP.Checked)
            {
                pOpenFromFile.Enabled = false;
                pOpenFromHTTP.Enabled = true;
                pOpenFromFTP.Enabled = false;
            }
            else if (rbOpenFromFTP.Checked)
            {
                pOpenFromFile.Enabled = false;
                pOpenFromHTTP.Enabled = false;
                pOpenFromFTP.Enabled = true;
            }
            else if (rbOpenFromClipboard.Checked)
            {
                pOpenFromFile.Enabled = false;
                pOpenFromHTTP.Enabled = false;
                pOpenFromFTP.Enabled = false;
            }
        }

        private void frmOpen_Load(object sender, EventArgs e)
        {
            if (Clipboard.ContainsImage())
            {
                rbOpenFromClipboard.Checked = true;
            }
            else
            {
                rbOpenFromFile.Checked = true;
            }
        }

        private void rbOpenFromFile_CheckedChanged(object sender, EventArgs e)
        {
            if (rbOpenFromFile.Checked)
            {
                UpdateUi();
            }
        }

        private void rbOpenFromFTP_CheckedChanged(object sender, EventArgs e)
        {
            if (rbOpenFromFTP.Checked)
            {
                UpdateUi();
            }
        }

        private void rbOpenFromHTTP_CheckedChanged(object sender, EventArgs e)
        {
            if (rbOpenFromHTTP.Checked)
            {
                UpdateUi();
            }
        }

        private void rbOpenFromClipboard_CheckedChanged(object sender, EventArgs e)
        {
            if (rbOpenFromClipboard.Checked)
            {
                UpdateUi();
            }
        }

        private void btnOpen_Click(object sender, EventArgs e)
        {
            bool ok = false;
            if (rbOpenFromFile.Checked)
            {
                if (!string.IsNullOrWhiteSpace(tbFile.Text) && File.Exists(tbFile.Text))
                {
                    GdPictureStatus status = _owner.DisplayFromFile(tbFile.Text);
                    if (status == GdPictureStatus.OK)
                    {
                        ok = true;
                    }
                    else
                    {
                        MessageBox.Show("Error during file loading: " + status.ToString(), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    }
                }
                else
                {
                    MessageBox.Show("You must select a file", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    return;
                }
            }
            else if (rbOpenFromFTP.Checked)
            {
                int port = 0;
                if (!string.IsNullOrWhiteSpace(tbHost.Text) && !string.IsNullOrWhiteSpace(tbPort.Text)
                    && int.TryParse(tbPort.Text, out port) && !string.IsNullOrWhiteSpace(tbPath.Text))
                {
                    GdPictureStatus status = _owner.DisplayFromFTP(tbHost.Text, tbPath.Text, tbLogin.Text, tbPassword.Text, port);
                    if (status == GdPictureStatus.OK)
                    {
                        ok = true;
                    }
                    else
                    {
                        MessageBox.Show("FTP Error: " + _owner.GetLastTransferError(), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    }
                }
                else
                {
                    MessageBox.Show("You must enter all data for FTP connection", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            else if (rbOpenFromHTTP.Checked)
            {
                Uri uri = null;
                if (!string.IsNullOrWhiteSpace(tbOpenFromHTTP.Text) && Uri.TryCreate(tbOpenFromHTTP.Text, UriKind.Absolute, out uri))
                {
                    GdPictureStatus status = _owner.DisplayFromHTTP(uri.ToString(), "", "");
                    if (status == GdPictureStatus.OK)
                    {
                        ok = true;
                    }
                    else
                    {
                        MessageBox.Show("HTTP Error: " + _owner.GetLastTransferError(), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    }
                }
                else
                {
                    MessageBox.Show("You must specify an URI", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            else if (rbOpenFromClipboard.Checked)
            {
                if (Clipboard.ContainsImage())
                {
                    GdPictureStatus status = _owner.DisplayFromClipboard();
                    if (status == GdPictureStatus.OK)
                    {
                        ok = true;
                    }
                    else
                    {
                        MessageBox.Show("Error: " + status.ToString(), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    }
                }
                else
                {
                    MessageBox.Show("There is no image in clipboard", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            if (ok)
            {
                Close();
            }
        }

        private void btOpenFromFile_Click(object sender, EventArgs e)
        {
            DialogResult result = OpenFileDialog1.ShowDialog();

            if (result == DialogResult.OK)
            {
                tbFile.Text = OpenFileDialog1.FileName;
            }
        }
    }
}

File: GdPicture/WinForm/vb.net/Pages Manipulation/frmOpen.vb
Option Strict On
Option Explicit On

Imports System.IO
Imports GdPicture14

Public Class frmOpen

    Private ReadOnly m_owner As GdViewer

    Public Sub New(ByVal owner As GdViewer)
        InitializeComponent()
        m_owner = owner
    End Sub

    Private Sub UpdateUi()
        If rbOpenFromFile.Checked Then
            pOpenFromFile.Enabled = True
            pOpenFromHTTP.Enabled = False
            pOpenFromFTP.Enabled = False
        ElseIf rbOpenFromHTTP.Checked Then
            pOpenFromFile.Enabled = False
            pOpenFromHTTP.Enabled = True
            pOpenFromFTP.Enabled = False
        ElseIf rbOpenFromFTP.Checked Then
            pOpenFromFile.Enabled = False
            pOpenFromHTTP.Enabled = False
            pOpenFromFTP.Enabled = True
        ElseIf rbOpenFromClipboard.Checked Then
            pOpenFromFile.Enabled = False
            pOpenFromHTTP.Enabled = False
            pOpenFromFTP.Enabled = False
        End If
    End Sub

    Private Sub frmOpen_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Clipboard.ContainsImage() Then
            rbOpenFromClipboard.Checked = True
        Else
            rbOpenFromFile.Checked = True
        End If
    End Sub

    Private Sub rbOpenFromFile_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rbOpenFromFile.CheckedChanged
        If rbOpenFromFile.Checked Then
            updateUI()
        End If
    End Sub

    Private Sub rbOpenFromFTP_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rbOpenFromFTP.CheckedChanged
        If rbOpenFromFTP.Checked Then
            updateUI()
        End If
    End Sub

    Private Sub rbOpenFromHTTP_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rbOpenFromHTTP.CheckedChanged
        If rbOpenFromHTTP.Checked Then
            updateUI()
        End If
    End Sub

    Private Sub rbOpenFromClipboard_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rbOpenFromClipboard.CheckedChanged
        If rbOpenFromClipboard.Checked Then
            updateUI()
        End If
    End Sub

    Private Sub btnOpen_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOpen.Click
        Dim ok As Boolean = False
        If rbOpenFromFile.Checked Then
            If Not String.IsNullOrWhiteSpace(tbFile.Text) And File.Exists(tbFile.Text) Then
                Dim status As GdPictureStatus = m_owner.DisplayFromFile(tbFile.Text)
                If status = GdPictureStatus.OK Then
                    ok = True
                Else
                    MessageBox.Show("Error during file loading: " + status.ToString(), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
                End If
            Else
                MessageBox.Show("You must select a file", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
                Return
            End If
        ElseIf rbOpenFromFTP.Checked Then
            Dim port As Integer
            If Not String.IsNullOrWhiteSpace(tbHost.Text) And Not String.IsNullOrWhiteSpace(tbPort.Text) _
                And Integer.TryParse(tbPort.Text, port) And Not String.IsNullOrWhiteSpace(tbPath.Text) Then
                Dim status As GdPictureStatus = m_owner.DisplayFromFTP(tbHost.Text, tbPath.Text, tbLogin.Text, tbPassword.Text, port)
                If status = GdPictureStatus.OK Then
                    ok = True
                Else
                    MessageBox.Show("FTP Error: " + m_owner.GetLastTransferError(), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
                End If
            Else
                MessageBox.Show("You must enter all data for FTP connection", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
            End If
        ElseIf rbOpenFromHTTP.Checked Then
            Dim uri As Uri = Nothing
            If Not String.IsNullOrWhiteSpace(tbOpenFromHTTP.Text) And System.Uri.TryCreate(tbOpenFromHTTP.Text, UriKind.Absolute, uri) Then
                Dim status As GdPictureStatus = m_owner.DisplayFromHTTP(uri.ToString(), "", "")
                If status = GdPictureStatus.OK Then
                    ok = True
                Else
                    MessageBox.Show("HTTP Error: " + m_owner.GetLastTransferError(), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
                End If
            Else
                MessageBox.Show("You must specify an URI", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
            End If
        ElseIf rbOpenFromClipboard.Checked Then
            If Clipboard.ContainsImage() Then
                Dim status As GdPictureStatus = m_owner.DisplayFromClipboard()
                If status = GdPictureStatus.OK Then
                    ok = True
                Else
                    MessageBox.Show("Error: " + status.ToString(), "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
                End If
            Else
                MessageBox.Show("There is no image in clipboard", "Error", MessageBoxButtons.OK, MessageBoxIcon.Error)
            End If
        End If
        If ok Then
            Me.Close()
        End If
    End Sub

    Private Sub btOpenFromFile_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btOpenFromFile.Click
        Dim result As DialogResult = OpenFileDialog1.ShowDialog()

        If (result = DialogResult.OK) Then
            tbFile.Text = OpenFileDialog1.FileName
        End If
    End Sub
End Class